build/Dockerfile

```diff
@@ -1,27 +1,30 @@
-FROM golang:1.21 AS go_builder
+FROM --platform=${BUILDPLATFORM} golang:1.21 AS go_builder
 WORKDIR /go/src/github.com/percona/percona-server-mongodb-operator
 
-COPY . .
+COPY go.mod go.sum ./
+RUN go mod download
 
 ARG GIT_COMMIT
 ARG GIT_BRANCH
 ARG GO_LDFLAGS
 ARG GOOS=linux
+ARG TARGETARCH
```

Comment:
The same it was removed due to multi-arch builds  as I remember golang:1.21 has this ver my default and build is correct without it.

Reply:
JFI I have updated https://github.com/percona/percona-server-mysql-operator/blob/main/e2e-tests/build please check and copy changes

Reply:
Hello, TARGETARCH BUILDPLATFORM are buildx system variables https://docs.docker.com/build/guide/multi-platform/

TARGETARCH has the same value as golang (amd64/arm64)

I am using the same technic in others projects.

Reply:
I see, thanks.